py/redrock/fitz.py

```diff
@@ -107,7 +107,25 @@ def minfit(x, y):
     return (x0, xerr, y0, zwarn)
 
 
-def fitz(zchi2, redshifts, target, template, nminima=3, archetype=None, use_gpu=False):
+
+def legendre_calculate(nleg, dwave):
+    wave = np.concatenate([ w for w in dwave.values() ])
+    wmin = wave.min()
+    wmax = wave.max()
+    legendre = { hs:np.array([scipy.special.legendre(i)( (w-wmin)/(wmax-wmin)*2.) for i in range(nleg)]) for hs, w in dwave.items() }
+
+    return legendre
+
+def prior_on_coeffs(n_nbh, deg_legendre, sigma):
+
+    nbasis = n_nbh+deg_legendre*3 # 3 desi cameras
```

Comment:
Please explore how to avoid hardcoding 3 as the number of DESI cameras, e.g. by inspecting the number of different wavelength grids represented in the input spectra.  Redrock has been used by eBOSS (with 2 cameras coadded into 1 before fitting), and by WEAVE (2 cameras), and could also be used on DESI with coadded cameras.  Let's try to preserve that generality.

Reply:
I think, we can think of adding ncam in rrdesi. Fixing the default value to 3.

Reply:
Fixed now, code can calculate number of camera using spectra information